compiler/optimizer/VPHandlers.cpp

```diff
@@ -2150,6 +2150,57 @@ TR::Node *constrainIaload(OMR::ValuePropagation *vp, TR::Node *node)
 
          node->setIsNonNull(true);
          }
+
+      else if (base
+               && base->getKnownObject()
+               && !symRef->isUnresolved() && symRef->getSymbol()->isRecognizedKnownObjectShadow()
+               && !vp->comp()->compileRelocatableCode())
+         // Base is known object, and field is recognized known object shadow,
+         // hence can be treated as known object.
+         {
+         TR::VMAccessCriticalSection constrainIaloadCriticalSection(vp->comp(),
+               TR::VMAccessCriticalSection::tryToAcquireVMAccess);
+         if (constrainIaloadCriticalSection.hasVMAccess())
+            {
+            TR::KnownObjectTable *knot = vp->comp()->getOrCreateKnownObjectTable();
+            uintptrj_t baseObject = knot->getPointer(base->getKnownObject()->getIndex());
+            // Check if baseObject is actually the recognized field Class
+            TR_OpaqueClassBlock *baseObjectClazz = TR::Compiler->cls.objectClass(vp->comp(), baseObject);
```

Comment:
We should check before loading the class pointer that `baseObject != 0`. VP tries not to create known object constraints for null, but there is a null known object index.

Reply:
Added null check